overlord/devicestate/devicemgr.go

```diff
@@ -1543,6 +1543,56 @@ func (m *DeviceManager) ensurePostFactoryReset() error {
 	return os.Remove(factoryResetMarker)
 }
 
+func (m *DeviceManager) hasUserExpired(user *auth.UserState) bool {
+	// If the user has no expiration date, then Expiration should not
+	// be set, and contain the default value.
+	if user.Expiration.IsZero() {
+		return false
+	}
+
+	return user.Expiration.Before(time.Now())
+}
+
+// ensureExpiredUsersRemoved is periodically called as a part of Ensure()
+// to remove expired users from the system.
+func (m *DeviceManager) ensureExpiredUsersRemoved() error {
+	st := m.state
+	st.Lock()
+	defer st.Unlock()
+
+	// So far this is only set to be done in run mode, it might not
+	// make any sense to do in it any other mode.
+	mode := m.SystemMode(SysHasModeenv)
```

Comment:
this should use SysAny if the intention is really to work the same on also classic systems

Reply:
Thanks for this, I changed this but I don't think I understand the Modeenv correctly - Sent you a DM.